### PR TITLE
2ai documentation improvement

### LINKED ISSUE #96: Create Developer Documentation (Express)
Same as in Enketo Core: https://github.com/enketo/enketo-core/issues/580

### PR DESCRIPTION
**Pull request by magicznyleszek**
I did similar changes that `enketo-core` has :) Had to fix some comments and add a few for generator to understand the files. Also had to move some stuff around and split/order things by how I think they should be - please check if these are ok.

Closes #96

### CODE REVIEW

File: app/models/config-model.js [hunk 1]
```diff
@@ -1,3 +1,7 @@
+/**
+ * @module confif-model
```
Comment (MartijnR) [inline app/models/config-model.js]: typo

File: tutorials/10-configuration.md [hunk 2]
```diff
@@ -1,11 +1,11 @@
 Configuration
 ==============
 
-All configuration is done in config/config.json or with equivalent environment variables (see [sample.env](./sample.env)). **Leave config/default-config.json unchanged.** Whichever of these 2 methods you choose, will override the defaults set in config/default-config.json.
+All configuration is done in config/config.json or with equivalent environment variables (see [sample.env](https://github.com/enketo/enketo-express/blob/master/config/sample.env)). **Leave config/default-config.json unchanged.** Whichever of these 2 methods you choose, will override the defaults set in config/default-config.json.
 
 Below is a complete list of all configuration items. The **bold items are important to set**. Others are less important.
 
-#### app name 
+#### app name
```
Comment (MartijnR) [inline tutorials/10-configuration.md]: Can these configuration items (or just all ### headers) get a bookmark (i.e. a id attribute)? We're often pointing users to the documentation of a particular config item. For this one `id="app-name"` as in https://github.com/enketo/enketo-express/blob/master/config/README.md#app-name

**Comment by MartijnR** [comment]
The travis error is caused by running jsbeatifier on the `docs` folder.

File: tutorials/10-configuration.md (hunk 2 above)
Comment (magicznyleszek) [reply tutorials/10-configuration.md]: You can do it with this JSDoc already - all the headers automatically get `id`s (e.g. https://enketo.github.io/enketo-core/tutorial-20-development.html#notes-for-css-developers). The only problem is that you need to get it from inspecting the code, as there are no fancy anchor icons as you usually have.

File: .publishrc [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    "validations": {
+        "vulnerableDependencies": false,
+        "uncommittedChanges": true,
+        "untrackedFiles": true,
+        "sensitiveData": false,
+        "branch": "master",
+        "gitTag": true
+    },
+    "confirm": true,
+    "publishCommand": "npm publish",
```
[the remark this answers is not in the document]
Comment (magicznyleszek) [reply .publishrc]: Right! I removed all publishin related things.

File: app/models/config-model.js (hunk 1 above)
Comment (magicznyleszek) [reply app/models/config-model.js]: Fixed

**Comment by magicznyleszek** [comment]
Fixed the travis errors 🖖

File: tutorials/10-configuration.md (hunk 2 above)
Comment (MartijnR) [reply tutorials/10-configuration.md]: Ah right. That will do for now!

**Comment by MartijnR** [comment]
Great. Will merge. Just for my interest why did you choose to include `rimraf` instead of using `rm -rf`?

**Comment by MartijnR** [comment]
Ah I guess for cross-platform use (Windows)

**Pull request merged**